server/src/main/java/org/apache/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -282,6 +282,13 @@ public void stop()
   public void add(String lookupName, LookupExtractorFactoryContainer lookupExtractorFactoryContainer)
   {
     Preconditions.checkState(lifecycleLock.awaitStarted(1, TimeUnit.MILLISECONDS));
+    LookupLoadingSpec lookupLoadingSpec = lookupListeningAnnouncerConfig.getLookupLoadingSpec();
+    if (lookupLoadingSpec.getMode() == LookupLoadingSpec.Mode.NONE ||
+        (lookupLoadingSpec.getMode() == LookupLoadingSpec.Mode.ONLY_REQUIRED
+         && !lookupLoadingSpec.getLookupsToLoad().contains(lookupName))) {
+      LOG.warn("Skipping notice to add lookup [%s] : current lookup loading mode does not allow it.", lookupName);
```

Comment:
a drive-by-comment: an `info` log would be more appropriate here, since this is expected behavior when the user sets it.
```suggestion
      LOG.info("Skipping notice to add lookup[%s] since current lookup loading mode[%s] does not allow it.", lookupName, lookupLoadingSpec.getMode());
```